File: Domains/InfiniteMario/15-rl-competition-2009/system/codecs/Python/src/tests/test_empty_environment.py
# 
# Copyright (C) 2008, Brian Tanner
# 
#http://rl-glue-ext.googlecode.com/
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
#  $Revision: 446 $
#  $Date: 2009-01-22 22:20:21 -0500 (Thu, 22 Jan 2009) $
#  $Author: brian@tannerpages.com $
#  $HeadURL: http://rl-glue-ext.googlecode.com/svn/trunk/projects/codecs/Python/src/tests/test_empty_environment.py $

import random
import sys
from rlglue.environment.Environment import Environment
from rlglue.environment import EnvironmentLoader as EnvironmentLoader
from rlglue.types import Observation
from rlglue.types import Action
from rlglue.types import Reward_observation_terminal

class test_empty_environment(Environment):
	whichEpisode=0
	emptyObservation=Observation()
	nonEmptyObservation=Observation(2,4,5)

	def env_init(self):  
		self.nonEmptyObservation.intArray=[0,1]
		self.nonEmptyObservation.doubleArray=[0.0/4.0,1.0/4.0,2.0/4.0,3.0/4.0]
		self.nonEmptyObservation.charArray=['a','b','c','d','e']
		return ""

	def env_start(self):
		self.whichEpisode=self.whichEpisode+1
		
		
		if self.whichEpisode % 2 == 0:
			return self.emptyObservation
		else:
			return self.nonEmptyObservation
	
	def env_step(self,action):
		ro=Reward_observation_terminal()
		
		if self.whichEpisode % 2 == 0:
			ro.o=self.emptyObservation
		else:
			ro.o=self.nonEmptyObservation

		return ro	

	def env_cleanup(self):
		pass

	def env_message(self,inMessage):
		return None
	
if __name__=="__main__":
	EnvironmentLoader.loadEnvironment(test_empty_environment())

File: Domains/InfiniteMario/15-rl-competition-2009/system/codecs/Matlab/src/tests/test_empty_environment.m
%  Copyright 2008 Brian Tanner
%  http://rl-glue-ext.googlecode.com/
%  brian@tannerpages.com
%  http://research.tannerpages.com
%  
%   Licensed under the Apache License, Version 2.0 (the "License");
%  you may not use this file except in compliance with the License.
%   You may obtain a copy of the License at
%  
%       http://www.apache.org/licenses/LICENSE-2.0
%  
%   Unless required by applicable law or agreed to in writing, software
%   distributed under the License is distributed on an "AS IS" BASIS,
%   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
%   See the License for the specific language governing permissions and
%   limitations under the License.
%  
%   $Revision: 637 $
%   $Date: 2009-02-07 16:02:45 -0500 (Sat, 07 Feb 2009) $
%   $Author: brian@tannerpages.com $
%  $HeadURL: http://rl-glue-ext.googlecode.com/svn/trunk/projects/codecs/Matlab/src/tests/test_empty_environment.m $
%
function theEnvironment=test_empty_environment()
    theEnvironment.env_init=@test_empty_environment_init;
    theEnvironment.env_start=@test_empty_environment_start;
    theEnvironment.env_step=@test_empty_environment_step;
    theEnvironment.env_cleanup=@test_empty_environment_cleanup;
    theEnvironment.env_message=@test_empty_environment_message;
end

function taskSpec=test_empty_environment_init()
    global test_empty_environment_struct;

	test_empty_environment_struct.whichEpisode=0;
	test_empty_environment_struct.emptyObservation=org.rlcommunity.rlglue.codec.types.Observation();
	test_empty_environment_struct.nonEmptyObservation=org.rlcommunity.rlglue.codec.types.Observation();
	test_empty_environment_struct.nonEmptyObservation.intArray=[0 1];
	test_empty_environment_struct.nonEmptyObservation.doubleArray=[0/4 1/4 2/4 3/4];
	test_empty_environment_struct.nonEmptyObservation.charArray=['abcde'];

	taskSpec='';
end    


function theObservation=test_empty_environment_start()
    global test_empty_environment_struct;

	test_empty_environment_struct.whichEpisode=test_empty_environment_struct.whichEpisode+1;

	if mod(test_empty_environment_struct.whichEpisode,2) == 0
		theObservation=test_empty_environment_struct.emptyObservation;
	else
		theObservation=test_empty_environment_struct.nonEmptyObservation;
	end
end


function rewardObservation=test_empty_environment_step(theAction)
    global test_empty_environment_struct;

	rewardObservation=org.rlcommunity.rlglue.codec.types.Reward_observation_terminal();

	if mod(test_empty_environment_struct.whichEpisode,2) == 0
		rewardObservation.o=test_empty_environment_struct.emptyObservation;
	else
		rewardObservation.o=test_empty_environment_struct.nonEmptyObservation;
	end
end

function returnMessage=test_empty_environment_message(theMessageJavaObject)
	returnMessage='';
end

function test_empty_environment_cleanup()
end
